java/code/src/com/redhat/rhn/domain/channel/ChannelFactory.java

```diff
@@ -1337,7 +1337,7 @@ public static void addErrataToChannel(Set<Long> eids, Long cid) {
     public static ContentSource findVendorContentSourceByRepo(String repoUrl) {
         Criteria criteria = getSession().createCriteria(ContentSource.class);
         criteria.add(Restrictions.isNull("org"));
-        if (repoUrl.contains("mirrorlist.centos.org")) {
+        if (repoUrl.contains("mirrorlist.centos.org") || repoUrl.contains("mirrors.rockylinux.org")) {
```

Comment:
My understanding his that this is needed if the URLs are mirrorlists and not repositories. This is the case for RockyLinux.